src/entities/TextNode/index.ts

```diff
@@ -1,43 +1,194 @@
+  public get serialized(): InlineNodeSerialized {
+    return {
+      text: this.getText(),
+      // No fragments for text node
+      fragments: [],
+    };
+  }
 
   /**
-   * Constructor for TextNode class
+   * Inserts text to specified position. By default, appends new text to the current value
    *
-   * @param args - TextNode constructor arguments.
-   * @param args.value - Text content of the node.
-   * @param args.children - Array of child nodes.
-   * @param args.parent - A parent of TextNode.
+   * @param text - text to insert
+   * @param [index] - char start index
    */
-  constructor({ value, children = [], parent }: TextNodeConstructorParameters) {
-    this.#value = value;
-    this.#length = value.length;
-    this.#children = children;
-    this.#parent = parent;
+  public insertText(text: string, index = this.length): void {
+    this.#validateIndex(index);
+
+    this.#value = this.#value.slice(0, index) + text + this.#value.slice(index);
+  }
+
+  /**
+   * Remove text from specified range
+   *
+   * @param [start] - start char index of the range, 0 by default
+   * @param [end] - end char index of the range, text length by default
+   * @returns {string} removed text
+   */
+  public removeText(start = 0, end = this.length): string {
+    this.#validateIndex(start);
+    this.#validateIndex(end);
+
+    const removedValue = this.#value.slice(start, end);
+
+    this.#value = this.#value.slice(0, start) + this.#value.slice(end);
+
+    if (this.length === 0) {
+      this.remove();
+    }
+
+    return removedValue;
+  }
+
+  /**
+   * Returns text value from the specified range
+   *
+   * @param [start] - start char index of the range, 0 by default
+   * @param [end] - end char index of the range, text length by default
+   */
+  public getText(start = 0, end = this.length): string {
+    if (start > end) {
+      // Stryker disable next-line StringLiteral
+      throw new Error(`Start index ${start} should be less or equal than end index ${end}`);
+    }
+
+    this.#validateIndex(start);
+    this.#validateIndex(end);
+
+    return this.#value.slice(start, end);
+  }
+
+  /**
+   * Applies inline tool for specified range
+   *
+   * @param tool - name of the tool to apply
+   * @param start - start char index of the range
+   * @param end - end char index of the range
+   * @param [data] - inline tool data if applicable
+   * @returns {InlineNode[]} - array of nodes after applied formatting
+   */
+  public format(tool: InlineToolName, start: number, end: number, data?: InlineToolData): InlineNode[] {
+    this.#validateIndex(start);
+    this.#validateIndex(end);
+
+    const formattingNode = new FormattingNode({
+      tool,
+      data,
+    });
+
+    const fragments: ChildNode[] = [];
+
+    /**
+     * If start index is greater than 0, we need extract part of the text before the start index
+     */
+    if (start > 0) {
+      fragments.push(this.#cloneFragment(0, start));
```

Comment:
I think, it's better to rename `cloneFragment` to `extractFragment` or `createFragment` or `composeFragment`

Reply:
I think `clone` is more accurate, but I've replaced `fragment` to `contents` as we usually refer to fragment in different context

Reply:
`clone` means making a copy, but this method created a different text node with different offsets.